system-addon/lib/TopSitesFeed.jsm

```diff
@@ -57,6 +65,16 @@ this.TopSitesFeed = class TopSitesFeed {
     }
   }
 
+  filterForThumbnailExpiration(callback) {
+    const {initialized, rows} = this.store.getState().TopSites;
+
+    if (initialized) {
+      callback(rows.map(site => site.url));
+    } else {
+      callback([]);
```

Comment:
nit: TopSites state should always has an array for rows, so the condition is unnecessary